pkg/cloud/amazon/common.go

```diff
@@ -69,10 +69,17 @@ func ResolveRegionWithoutOptions() (string, error) {
 func ParseContext(context string) (string, string, error) {
 	reg := regexp.MustCompile(`([a-zA-Z][-a-zA-Z0-9]*)\.((us(-gov)?|ap|ca|cn|eu|sa)-(central|(north|south)?(east|west)?)-\d)\.*`)
 	result := reg.FindStringSubmatch(context)
-	if len(result) < 3 {
-		return "", "", errors.Errorf("unable to parse %s as <cluster_name>.<region>.*", context)
+	// Also check if the context name matchesAWS ARN format as defined:
+	// https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonelasticcontainerserviceforkubernetes.html#amazonelasticcontainerserviceforkubernetes-resources-for-iam-policies
+	arnReg := regexp.MustCompile(`arn:aws:eks:((?:us(?:-gov)?|ap|ca|cn|eu|sa)-(?:central|(?:north|south)?(?:east|west)?)-\d):[0-9]*:cluster\/([a-zA-Z][-a-zA-Z0-9]*)`)
+	arnResult := arnReg.FindStringSubmatch(context)
+	if len(result) >= 3 {
```

Comment:
I find the logic a bit odd here. We are creating and matching both regexp every time regardless. I would more expect that we test one regex (`reg` since it is the default for when we create the context) and if it matches we return directly. If there is no match, we try the next.

One could just move this if statement up above the creation and match of `arnReg`.

That's for the case where we stick to this parsing of the context name. Would much prefer a more reliable approach.